ompi/mca/mtl/psm2/mtl_psm2_component.c

```diff
@@ -387,6 +372,26 @@ ompi_mtl_psm2_component_init(bool enable_progress_threads,
         ompi_mtl_psm2_set_shadow_env (ompi_mtl_psm2_shadow_variables + i);
     }
 
+#if OPAL_CUDA_SUPPORT
+    /*
+     * If using CUDA enabled OpenMPI, the user likely intends to
+     * run with CUDA buffers. So, force-set the envvar here if user failed
+     * to set it.
+     */
+    cuda_env = getenv("PSM2_CUDA");
+    if (!cuda_env) {
+        opal_show_help("help-mtl-psm2.txt",
+                       "no psm2 cuda env", true,
+                       ompi_process_info.nodename);
+        opal_setenv("PSM2_CUDA", "1", false, &environ);
+        cuda_envvar_set = 1;
+    } else if (strcmp(cuda_env, "0") == 0) {
+        opal_show_help("help-mtl-psm2.txt",
+                       "psm2 cuda env zero", true,
+                       ompi_process_info.nodename);
```

Comment:
So you're showing a help message:

* If `PSM2_CUDA` was not set
* If `PSM2_CUDA` was set to 0

The 2nd message there is... a bit weird.  So even if I (a user) am doing what I am supposed to be doing (i.e., telling you that I am not using CUDA buffers in my application), you're going to emit a warning message.  If you really want to do that, feel free -- this is Intel's plugin.  But I think that's downright weird.

Reply:
Intent was to inform user that _only_ host buffers will work with `PSM2_CUDA=0`. I'll remove it. Should be reasonable to assume user will already know that.